app/assets/javascripts/discourse/components/search-advanced-options.js.es6

```diff
@@ -74,107 +74,175 @@ export default Em.Component.extend({
   },
 
   _update() {
-    let searchTerm = this.get('searchTerm');
-
-    if (!searchTerm) {
+    if (!this.get('searchTerm')) {
       this._init();
       return;
     }
 
-    this.findUsername(searchTerm);
-    this.findCategory(searchTerm);
-    this.findGroup(searchTerm);
-    this.findBadge(searchTerm);
-    this.findTags(searchTerm);
-    this.findIn(searchTerm);
-    this.findStatus(searchTerm);
-    this.findPostsCount(searchTerm);
-    this.findPostTime(searchTerm);
+    this.setSearchedTermValue('searchedTerms.username', REGEXP_USERNAME_PREFIX);
+    this.setSearchedTermValueForCategory();
+    this.setSearchedTermValueForGroup();
+    this.setSearchedTermValueForBadge();
+    this.setSearchedTermValueForTags();
+    this.setSearchedTermValue('searchedTerms.in', REGEXP_IN_PREFIX);
+    this.setSearchedTermValue('searchedTerms.status', REGEXP_STATUS_PREFIX);
+    this.setSearchedTermValueForPostTime();
+    this.setSearchedTermValue('searchedTerms.posts_count', REGEXP_POST_COUNT_PREFIX);
   },
 
-  findSearchTerm(EXPRESSION, searchTerm) {
+  findSearchTerm(EXPRESSION) {
+    const searchTerm = this.get('searchTerm');
     if (!searchTerm)
-      return "";
+      return [];
+
+    let result = [];
+    const blocks = searchTerm.match(REGEXP_BLOCKS);
+    _.each(blocks, function(block) {
+      if (block.match(EXPRESSION))
+        result.push(block);
+    });
 
-    const expressionPosition = searchTerm.search(EXPRESSION);
-    if (expressionPosition === -1)
-      return "";
+    return result;
+  },
 
-    const remainingPhrases = searchTerm.substring(expressionPosition + 2);
-    let nextExpressionPosition = remainingPhrases.search(REGEXP_FILTER_PREFIXES);
-    if (nextExpressionPosition === -1)
-      nextExpressionPosition = remainingPhrases.length;
+  setSearchedTermValue(key, regexPrefix) {
+    const match = this.findSearchTerm(regexPrefix);
 
-    return searchTerm.substring(expressionPosition, nextExpressionPosition + expressionPosition + 2).trim().split(' ')[0];
+    if (match.length !== 0) {
+      const userInput = match[0].replace(regexPrefix, '');
+      if (this.get(key) !== userInput) {
+        this.set(key, userInput);
+      }
+    } else {
+      this.set(key, null);
+    }
   },
 
-  findUsername(searchTerm) {
-    const match = this.findSearchTerm(REGEXP_USERNAME_PREFIX, searchTerm);
+  setSearchedTermValueForCategory() {
+    const match = this.findSearchTerm(REGEXP_CATEGORY_PREFIX);
+    if (match.length !== 0) {
+      const existingInput = _.isArray(this.get('searchedTerms.category')) ? this.get('searchedTerms.category')[0] : this.get('searchedTerms.category');
```

Comment:
IIRC, there is something bizarre with how the category-chooser and other select2 items set their values. It can become a string versus an array with one item. I'll try removing it, but that is why the badge and group logic had something like this too.